app/views/partials/layout-grid.html

```diff
@@ -0,0 +1,242 @@
+                ng-focus="setActiveRow(subItem)" ng-blur="activeRowOnBlur(subItem)">
+              <input ng-switch-when="DT" name="{{subItem.question}}" type="text" ng-model="subItem.value"
+                lf-date="dateOptions" placeholder="{{subItem._toolTip}}" ng-disabled="subItem._readOnly"
+                id="{{subItem._elementId}}"
+                aria-labelledby="{{lfData._horizontalTableInfo[item._horizontalTableId].columnHeaders[$index].id}}"
+                ng-focus="setActiveRow(subItem)" ng-blur="activeRowOnBlur(subItem)">
+              <!-- Gillardo boostrap datetime picker -->
+              <div ng-switch-when="DTM" class="lf-dtm-picker-block">
+                <input name="{{subItem.question}}" type="text" class="form-control" ng-model="subItem.value"
+                  datetime-picker="{{uibDateTimePickerFormat}}" alt-input-formats="uibDatePickerAltFormats"
+                  is-open="isOpen" enable-time="true" close-on-date-selection="true" button-bar="uibDtmPickerButtonBar"
+                  datepicker-options="uibDatePickerOptions" timepicker-options="uibTimePickerOptions"
+                  placeholder="{{subItem._toolTip}}" ng-disabled="subItem._readOnly" id="{{subItem._elementId}}"
+                  aria-labelledby="{{lfData._horizontalTableInfo[item._horizontalTableId].columnHeaders[$index].id}}"
+                  ng-focus="setActiveRow(subItem)" ng-blur="activeRowOnBlur(subItem); uibDtmPickerOnBlur('input')">
+                <button type="button" class="ui-datepicker-trigger" ng-click="openUibDtmPicker($event)"></button>
+              </div>
+              <textarea ng-switch-when="TX" name="{{subItem.question}}" ng-model="subItem.value"
+                ng-attr-placeholder="{{subItem._toolTip}}" ng-disabled="subItem._readOnly" id="{{subItem._elementId}}"
+                aria-labelledby="{{lfData._horizontalTableInfo[item._horizontalTableId].columnHeaders[$index].id}}"
+                ng-keyup="autoExpand($event)" rows="1" ng-focus="setActiveRow(subItem)"
+                ng-blur="activeRowOnBlur(subItem)"></textarea>
+              <lf-attachment ng-switch-when="attachment" item="subItem" create-attachment="createAttachment(item)"
+                labelled-by=lfData._horizontalTableInfo[item._horizontalTableId].columnHeaders[$index].id></lf-attachment>
+              <input ng-switch-default name="{{subItem.question}}" type="text" ng-model="subItem.value"
+                placeholder="{{subItem._toolTip}}" ng-disabled="subItem._readOnly" id="{{subItem._elementId}}"
+                aria-labelledby="{{lfData._horizontalTableInfo[item._horizontalTableId].columnHeaders[$index].id}}"
+                ng-focus="setActiveRow(subItem)" ng-blur="activeRowOnBlur(subItem)">
+
+            </div>
+          </ng-form>
+        </td>
+
+
+        <!-- last row data -->
+        <td ng-repeat="cell in subItem.items" ng-if="targetShown(cell) && !cell._isHiddenFromView"
```

Comment:
The "last row" comment seems misleading. This is really creating an additional column for each nested item, right? Are there ever situations where each item has a different number of nested items, resulting in different number of columns in each row?

Reply:
@semicolin  yes its creates additional column for each nested item . i haven't found such situation where each item has a different number of nested items  . and will remove that  misleading comment